api/datastore/sql/sql.go

```diff
@@ -667,6 +685,168 @@ func (ds *SQLStore) GetRoutesByApp(ctx context.Context, appID string, filter *mo
 	return res, nil
 }
 
+func (ds *SQLStore) PutFn(ctx context.Context, fn *models.Fn) (*models.Fn, error) {
+	err := ds.Tx(func(tx *sqlx.Tx) error {
+		query := tx.Rebind(fmt.Sprintf(`%s WHERE name=?`, fnSelector))
```

Comment:
This needs to be by either (appId, FnName) or just use FnID here. I'm in favour of just using FnID